### ISSUE TITLE
2nd level account name may no longer start with digits

### ISSUE THREAD

**Issue by blais** [beancount/beancount#290]
**[Original report](https://bitbucket.org/blais/beancount/issue/290) by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

The changes accepted last night result in:

```
ValueError: Invalid account name: Assets:99Test
```

Whereas `Assets:Vouchers:99Ranch` is accepted. It seems 2nd level account names may no longer start with digits.  I assume this change was not intentional.

Test case:

```
2010-01-01 open Assets:99Test
2010-01-01 open Assets:Vouchers:99Ranch
2010-01-01 open Equity:Opening-Balance

2018-03-26 * "Commodity after amount"
  Assets:99Test                 1 EUR
  Equity:Opening-Balance

2018-03-26 * "Commodity after amount"
  Assets:Vouchers:99Ranch       1 EUR
  Equity:Opening-Balance
```

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

This change is unintentional.
Thanks for filing the ticket, will fix shortly.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Found it, my bad (stupid mistake, regular expressions for validation were swapped and there was no corresponding test). Will submit ASAP.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #290: Validation of account name components was incorrectly swapped

**Comment by blais** [comment]
**Original comment by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

Thanks, works as expected now!
